File: solutions/dart/high-scores/1/lib/high_scores.dart
class HighScores {
   
   List <int> scores;
   int late = 0;
   List<int>  rever = [];
   List<int> orderedElements  = [];

   HighScores(this.scores) {
      late = scores[scores.length - 1];
      
      for (int i = 0 ;  i  <= scores.length - 1 ;  i++) orderedElements.add(scores[i]);


      orderedElements.sort();

      
      for (int i = orderedElements.length - 1;  i >= 0 ;  i--) rever.add(orderedElements[i]);
   }
   

   int personalBest () => orderedElements[orderedElements.length - 1];

   
   List <int> personalTopThree() {
      if (rever.length >= 3) return [rever[0], rever[1], rever[2]];
      else if (scores.length == 2) return [rever[0], rever[1]];
      else return [rever[0]];

   }

   int latest () => late;
    
}

File: solutions/javascript/high-scores/1/high-scores.js
export class HighScores {
    constructor(scores) {
        this.score = scores
        this.ordered = [...this.score]
        this.ordered.sort((a,b) => a - b)
        this.ordered.reverse()
    }

    get scores() {
        return this.score
    }
  
    get latest() {
        return this.ordered[2]
    }
  
    get personalBest() {
       return this.ordered[0]
    }
  
    get personalTopThree() {
       return [this.ordered[0], this.ordered[1], this.ordered[2]]
    }
}
  
